### PR TITLE
Removed deprecated torch.lstsq, replaced by torch.linalg.lstsq

### PR DESCRIPTION
**Pull request by raphaelattias**
As of September 22, torch.lstsq is deprecated (https://github.com/pytorch/pytorch/pull/70980) and has been replaced by torch.linalg.lstsq. This PR solves an issue for Pytorch version 1.13.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (andreped) [inline .vscode/launch.json:1]: This should not be included. Please, remove the .vscode/ directory in a new commit.

File: example.py [hunk 2]
```diff
@@ -21,8 +21,6 @@
 torch_normalizer = torchstain.normalizers.MacenkoNormalizer(backend='torch')
```
Comment (andreped) [inline example.py:21]: This example does not need to be adjusted. It is meant to showcase all backends (tf included). Hence, revert the changes in this file and make a new commit.

File: torchstain/torch/normalizers/macenko.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def __find_concentration(self, OD, HE):
         Y = OD.T
 
         # determine concentrations of the individual stains
```
Comment (andreped) [inline torchstain/torch/normalizers/macenko.py:51]: Note that I am not sure that doing this is backward compatible, as I believe `torch.linalg.lstsq` does not exist in older versions of torch. You will need to find a fix that works for all versions.

**Review by andreped — CHANGES_REQUESTED** [review]
See comments.

File: .vscode/launch.json (hunk 1 above)
Comment (raphaelattias) [reply .vscode/launch.json]: Fixed thank you

File: example.py (hunk 2 above)
Comment (raphaelattias) [reply example.py]: Fixed

File: torchstain/torch/normalizers/macenko.py (hunk 3 above)
Comment (raphaelattias) [reply torchstain/torch/normalizers/macenko.py:51]: By the Torch repo that change was made in Torch 1.9. I suggest bumping the minimum version of Torch required from 1.8 to 1.9 to assure backward compatibility:

https://github.com/pytorch/pytorch/blob/b0c86caa1d46a16195682e2afe5456f97265aa53/torch/_linalg_utils.py#L111

File: .github/workflows/tests_full.yml [hunk 4]
```diff
@@ -73,7 +73,7 @@ jobs:
       matrix:
         os: [ windows-2019, ubuntu-18.04, macos-11 ]
         python-version: [ 3.6, 3.7, 3.8, 3.9 ]
-        pytorch-version: [1.8.0, 1.9.0, 1.10.0, 1.11.0, 1.12.0]
+        pytorch-version: [1.9.0, 1.10.0, 1.11.0, 1.12.0, 1.13.0]
```
Comment (andreped) [inline .github/workflows/tests_full.yml:76]: Sneaky fix ;) But it would be better if we also supported older versions of pytorch (< 1.9).

**Review by andreped — CHANGES_REQUESTED** [review]

File: .github/workflows/tests_full.yml [hunk 5]
```diff
@@ -73,7 +73,7 @@ jobs:
       matrix:
         os: [ windows-2019, ubuntu-18.04, macos-11 ]
         python-version: [ 3.6, 3.7, 3.8, 3.9 ]
-        pytorch-version: [1.9.0, 1.10.0, 1.11.0, 1.12.0, 1.13.0]
+        pytorch-version: [1.8.0, 1.9.0, 1.10.0, 1.11.0, 1.12.0, 1.13.0]
```
Comment (andreped) [inline .github/workflows/tests_full.yml:76]: Great! Just handling new/older torch version in the repo that remains :]

File: torchstain/torch/normalizers/macenko.py [hunk 6]
```diff
@@ -49,7 +49,11 @@ def __find_concentration(self, OD, HE):
...
-        return torch.linalg.lstsq(HE, Y)[0]
+
+        if torch.__version__ >= (1,9,0):
```
Comment (andreped) [inline torchstain/torch/normalizers/macenko.py:53]: Great! It is a little suboptimal to have this if statement being called every single time this method is called. Might be better to do this only once in the __init__() or similar. And then, this method knows which lstsq to use, but we can fix that in the near future. LGTM.

**Review by andreped — APPROVED** [review]
LGTM! Thanks!

**Comment by andreped** [comment]
Note that, as I'm not a maintainer of this project, @carloalbertobarbano is needed to:
1) Approve running workflows such that we can verify that all unit tests are successful.
2) To accept the PR or not.

He will probably return from holiday anytime soon, @raphaelattias :]

File: torchstain/torch/normalizers/macenko.py (hunk 6 above)
Comment (carloalbertobarbano) [reply torchstain/torch/normalizers/macenko.py:53]: I agree, it should be a rather quick fix to do (in init or in the file header)

**Comment by carloalbertobarbano** [comment]
Hi @raphaelattias, I finally came back from vacation. Thanks for your contribution! I will happily merge it once you refactor the torch version check

**Comment by andreped** [comment]
Oh yeah, sure. I guess you will make that final change yourself in the development branch. Not sure @raphaelattias is available to make that fix right now.

**Comment by raphaelattias** [comment]
Hi @andreped thank you for the feedback, I am working on it :)

**Comment by carloalbertobarbano** [comment]
Thanks for the cooperation @raphaelattias
